testing/integration/otel_test.go

```diff
@@ -1648,21 +1648,13 @@ func AssertMapsEqual(t *testing.T, m1, m2 mapstr.M, ignoredFields []string, msg
 	flatM1 := m1.Flatten()
 	flatM2 := m2.Flatten()
 	for _, f := range ignoredFields {
-		hasKeyM1, _ := flatM1.HasKey(f)
-		hasKeyM2, _ := flatM2.HasKey(f)
-
-		if !hasKeyM1 && !hasKeyM2 {
-			assert.Failf(t, msg, "ignored field %q does not exist in either map, please remove it from the ignored fields", f)
-		}
-
```

Comment:
Why are we removing this? I thought it was here intentionally to ensure the ignored lists are trimmed whenever an element is not necessary,

Reply:
I'm not in favor of coupling this test to beats implementation details.  If beats stops producing a field that we ignore, that isn't a failure for this test.  This test exists to help us make sure the documents are the same.  Both documents not having a field is the same.

That being said, I'll put it back, and we can reevaluate if it becomes a burden.